Arelle/Arelle
# Search for pip-installed plugins when installing plugin imports

#### Reason for change
Pip-installed plugins were not working as expected when pip-installed plugins imported other plugins (either pip-installed or built-in).

#### Description of change
Refactor plugin discovery by implementing `EntryPointRef` which handles the discovery, storage, and search of references to both built-in and installed plugins.

Update `freshenModules` to reload renamed plugins using the same process as missing plugins.

Implement plugin "aliases" to ease transitioning of built-in plugins to installed plugins.
This involved separating out the parsing of `__pluginInfo__` dictionaries from the recursive `moduleInfo` generation, which also means the "discovery" of plugins involves reading each of the discovered plugins to parse the "name" and "aliases" fields.

UI update to "Manage plugins" > "Select" menu. Removed nesting UI in favor of simple list of discovered plugins. Renamed some built-in plugins for consistency. Removed the `validate` plugin which was only used to import *all* validation plugins at once.

![Screenshot 2023-11-15 at 12 13 28 PM](https://github.com/Arelle/Arelle/assets/105066394/f9594930-5f90-439f-9db0-42c271c16bda)

#### Steps to Test
For each of the following scenarios:
- Open the plugin Select menu (Help > Manage plug-ins > Select) and confirm options appear as expected.
- Run Arelle while invoking the associated plugin to confirm it's loaded correctly.

___
#### Default Setup
Keep all existing built-in plugins.
```
Edgar Renderer: not visible
Validate EFM: visible
```

___
#### Manually Installed Plugin
Install EdgarRenderer manually
```
Edgar Renderer: visible
Validate EFM: visible
```

___
#### Pip-installed Plugin
Pip-install [this branch](https://github.com/austinmatherne-wk/EdgarRenderer/tree/python-package) which provides `EdgarRenderer` and dependent plugins as entry points.
Delete `validate/EFM`, `validate/EFM-htm`, and `transforms` built-in plugin directories.
```
Edgar Renderer: visible
Validate EFM: visible
```

___
**review**:
@Arelle/arelle
